src/payment/models/bank-account/bank-account.service.ts

```diff
@@ -17,11 +21,50 @@ export class BankAccountService {
     });
 
     return (
-      bankAccounts.find((b) => b.user.id === userId) ?? (await this.createBankAccount(iban, userId, bankAccounts[0]))
+      bankAccounts.find((b) => b.user.id === userId) ??
+      (await this.createBankAccountInternal(iban, userId, bankAccounts[0]))
     );
   }
 
-  private async createBankAccount(iban: string, userId: number, copyFrom?: BankAccount): Promise<BankAccount> {
+  //
+
+  async getUserBankAccounts(userId: number): Promise<BankAccount[]> {
+    return this.bankAccountRepo.find({ user: { id: userId } });
+  }
+
+  async createBankAccount(userId: number, dto: CreateBankAccountDto): Promise<BankAccount> {
+    const bankAccounts = await this.bankAccountRepo.findOne({
+      where: { iban: dto.iban, user: { id: userId } },
+      relations: ['user'],
+    });
+    if (bankAccounts) throw new ForbiddenException('bankAccount already exists');
+
+    const bankAccount = await this.getBankAccount(dto.iban, userId);
+
+    bankAccount.user = { id: userId } as User;
+    bankAccount.preferredCurrency = { id: dto.preferredCurrency.id } as Fiat;
+    if (dto.label) bankAccount.label = dto.label;
+
+    return this.bankAccountRepo.save(bankAccount);
+  }
+
+  async updateBankAccount(userId: number, dto: UpdateBankAccountDto): Promise<BankAccount> {
+    let entity = await this.bankAccountRepo.findOne({
+      where: { iban: dto.iban, user: { id: userId } },
+      relations: ['user'],
+    });
+    if (!entity) throw new NotFoundException('bankAccount not found');
+
+    const update = this.bankAccountRepo.create(dto);
```

Comment:
We would need to check the currency. Maybe in a separate createEntity method (see e.g. StakingService)?